### DISCUSSION TITLE
Waking up PICO on incoming UART data

### DISCUSSION THREAD

**Discussion by knfevg** [micropython/micropython#16103] [RP2040 / Pico]
Hello everybody,
I'm using a PICO with a Simcom 7080 NBIOT model for developing a IOT solution. The main goal is for the unit to be super power conservative. For this I'm putting the modem into sleep mode and want to put the pico in sleep to. But, as soon as the modem receives something and pushes it to the PICO (Rx/Tx UART) I want the pico to wake up. Any ideas how to do that?

**Comment by ThinkTransit** [comment]
You can probably do that using the RI (Ring Indicator) pin on the modem and use this signal to wake the Pico. You could also use the falling edge of the UART line to wake the Pico however you would lose/miss some of the initial data.  

However I don't think the sleep current on the Pico is very good(~1mA) compared to other micros, esp32 (~10uA).

![image](https://github.com/user-attachments/assets/d856c9e4-67eb-4a06-9784-4b6ff144f7b9)

**Comment by robert-hh** [comment]
You can set up a Pin IRQ on a Pin connected to the UART. When the selected transition happens, lightsleep terminates. The handler for that IRQ can be a dummy, or it can disable itself. The Pin which you select for IRQ can be any GPIO. It may not be possible to use the UART RX pin. But you can use another pin and connect RX to that Pin as well. 
In any case, if UART RX data wakes up the Pico, you will loose the first byte(s) of the UART message.

**Reply by GitHubsSilverBullet** [reply]
Define an appropriate protocol with CRC and send the datablock twice. Done.
